flocker/provision/_effect.py

```diff
@@ -0,0 +1,56 @@
+import sys
+from characteristic import attributes
+from effect import (
+    sync_performer, Effect,
+    ComposedDispatcher, TypeDispatcher, base_dispatcher)
+from effect.do import do, do_return
+
+
+# This should be upstream.
```

Comment:
Given the current upstream conversation, I think that it is worth copying the minor fixes from that PR to this one.